frontend/src/metabase/search/containers/SearchApp.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+import querystring from "querystring";
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen, waitFor } from "__support__/ui";
+import SearchApp from "metabase/search/containers/SearchApp";
+import { Route } from "metabase/hoc/Title";
+import {
+  setupDatabasesEndpoints,
+  setupSearchEndpoints,
+} from "__support__/server-mocks";
+import {
+  createMockDatabase,
+  createMockSearchResult,
+} from "metabase-types/api/mocks";
+import { SearchResult } from "metabase-types/api";
+
+import { SearchFilters } from "metabase/search/types";
+
+// Mock PAGE_SIZE so we don't have to generate a ton of elements for the pagination test
+jest.mock("metabase/search/containers/constants", () => ({
+  PAGE_SIZE: 3,
+}));
+
+const TRANSLATED_SIDEBAR_NAMES: Record<string, string> = {
+  "All results": "All results",
+  collection: "Collections",
+  dashboard: "Dashboards",
+  database: "Databases",
+  dataset: "Models",
+  metric: "Metrics",
+  pulse: "Pulses",
+  segment: "Segments",
+  table: "Raw Tables",
+  card: "Questions",
+};
+
+const TEST_ITEMS: Partial<SearchResult>[] = [
+  { name: "Card", model: "card" },
+  { name: "Collection", model: "collection" },
+  { name: "Dashboard", model: "dashboard" },
+  { name: "Database", model: "database" },
+  { name: "Dataset", model: "dataset" },
+  { name: "Table", model: "table" },
+  { name: "Pulse", model: "pulse" },
+  { name: "Segment", model: "segment" },
+  { name: "Metric", model: "metric" },
+];
+
+const TEST_SEARCH_RESULTS: SearchResult[] = TEST_ITEMS.map((metadata, index) =>
+  createMockSearchResult({ ...metadata, id: index + 1 }),
+);
+
+const TEST_DATABASE = createMockDatabase();
+
+const setup = async ({
+  searchText,
+  searchFilters = {},
+  searchItems = TEST_SEARCH_RESULTS,
+}: {
+  searchText?: string;
+  searchFilters?: SearchFilters;
+  searchItems?: SearchResult[];
+}) => {
+  setupDatabasesEndpoints([TEST_DATABASE]);
+  setupSearchEndpoints(searchItems);
+
+  // for testing the hydration of search text and filters on page load
+  const params = {
+    ...searchFilters,
+    q: searchText,
+  };
+  const searchParams = querystring.stringify(params);
```

Comment:
FYI from https://nodejs.org/api/querystring.html

> `querystring` is more performant than [URLSearchParams](https://nodejs.org/api/url.html#class-urlsearchparams) but is not a standardized API. Use [URLSearchParams](https://nodejs.org/api/url.html#class-urlsearchparams) when performance is not critical or when compatibility with browser code is desirable.

Not a requested change, but I'd use `URLSearchParams` in this case since it's more standardized.